clash-prelude/src/Clash/Signal/Internal.hs

```diff
@@ -386,7 +386,7 @@ data SDomainConfiguration (dom :: Domain) (conf :: DomainConfiguration) where
 
 deriving instance Show (SDomainConfiguration dom conf)
 
-type KnownConfiguration dom conf = (KnownDomain dom, KnownConf dom ~ conf)
+type KnownConfiguration dom conf = (KnownDomain dom, KnownConf dom ~ conf, KnownNat (DomainConfigurationPeriod conf))
```

Comment:
Could we add this is a superclass of `KnownDomain`?

Reply:
No, this is the wrong approach to begin with. We should extract the SNat out of the domain configuration in the `trueDualPortRam` function instead of having the `KnownNat` constraint. i.e. we already have `KnownDomain` constraint, get the `SDomainConfiguration` out of that, and use the `SNat` inside of it.

Reply:
Agreed. @leonschoorl This works too:

```haskell
case knownDomain @someDom of
  SDomainConfiguration _ sPeriod _ _ _ _ _ _ _ _
```

Ideally this would be a record type, but I didn't know you could do that for GADTs when I implemented synthesis domains.

Reply:
Or is there another reason you opted for `KnowNat` instead?